### DISCUSSION TITLE
V7 inheritance demo gives Typescript error

### DISCUSSION THREAD

**Discussion by JamieS1211** [sequelize/sequelize#16953] [Help & Questions]
Hi there, I have just started migrating over to sequelize 7 and have run into an issue with the example for model inheritance. 

The documentation can be found here- https://sequelize.org/docs/v7/models/inheritance/

The example suggests for the abstract class to be:
```
@Table.Abstract
class Player<M extends Player = Player> extends Model<InferAttributes<M>, InferCreationAttributes<M>> {
  @Attribute(DataTypes.STRING)
  @NotNull
  declare name: string;
}
```

However when doing this I get a typescript error: 
```
Type parameter 'M' has a circular default.ts(2716)
⚠ Error(TS2716) | 
Type parameter M has a circular default.
class Player<M extends Player<unknown>>
```

If you have any advice here would be fantastic as I am totally stumped. I agree with your example but it IS circular and if thats an issue for typescript not sure what to do sort of "ts-ignore" 🤮

**Comment by hanrea** [comment]
I also encountered this problem 😞

**Comment by hanrea** [comment]
@JamieS1211 


```typescript
@Table.Abstract
export abstract class Player<M extends Player<M>> extends Model<InferAttributes<M>, InferCreationAttributes<M>> {
    @PrimaryKey
    @AutoIncrement
    @Attribute(DataTypes.INTEGER)
    declare id: CreationOptional<number>;
}

@Table.Abstract
export class User<M extends TenantBase<M>> extends Player<M> {
    @NotNull
    @Attribute(DataTypes.STRING)
    declare name: string; 
}

export class Test extends UserBase<Test> {
    @NotNull
    @Attribute(DataTypes.STRING)
    declare userinfo: string; 
}
```

**Reply by hanrea** [reply]
I successfully created the table using this method, but the field order is reversed.

like as :
`userinfo `  | `name` | `id`
